File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 5/10/2026, 11:51:08 PM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxcount = 0;
        int count = 0;
        for(int i=0; i<nums.size(); i++){
            if(nums[i] == 1){
                count++;
                maxcount = max(maxcount, count);
            }else{
                count = 0;
            }
        }
        return maxcount;
    }
};

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
// Last updated: 5/10/2026, 11:51:00 PM
import java.io.FileWriter;
import java.io.IOException;

public class Main {
    static {
        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            try (FileWriter fw = new FileWriter("display_runtime.txt")) {
                fw.write("0");
            } catch (IOException e) {
                e.printStackTrace();
            }
        }));
    }
}
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int count = 0;
        int maxi = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                count = 0;
            } else {
                count++;
            }
            maxi = Math.max(maxi, count);
        }

        return maxi;
    }
}
